scripts/zap-scanner.sh

```diff
@@ -37,60 +34,74 @@ elif [ "$ENVIRONMENT" = "local" ]; then
   -config spider.postform=true"
 
-echo "================== OWASP ZAP tests =================="
-cd $TARGET_DIR
-
-if [ "$TARGET" = "frontend" ]; then
-    docker-compose down
-    docker-compose up -d --build
-fi
-
-# Ensure the reports directory can be written to
-chmod 777 $(pwd)/reports
+# How long ZAP will crawl the app with the spider process
+ZAP_SPIDER_MINS=5
 
+ZAP_ARGS=(-t "$APP_URL" -m "$ZAP_SPIDER_MINS" -r "$REPORT_NAME" -z "$ZAP_CLI_OPTIONS")
 if [ -z ${CONFIG_FILE+x} ]; then
-    echo "No config file"
-    docker-compose run zaproxy zap-full-scan.py \
-                   -t $APP_URL \
-                   -m 5 \
-                   -z "${ZAP_CONFIG}" \
-                   -r "$REPORT_NAME" | tee /dev/tty | grep -q "FAIL-NEW: 0"
+    echo "No config file, defaulting all alerts to WARN"
 else
-    echo "Config file $ENVIRONMENT"
-    docker-compose run zaproxy zap-full-scan.py \
-                   -t $APP_URL \
-                   -m 5 \
-                   -z "${ZAP_CONFIG}" \
-                   -c "$CONFIG_FILE" \
-                   -r "$REPORT_NAME"  | tee /dev/tty | grep -q "FAIL-NEW: 0"
+    echo "Config file found"
+    ZAP_ARGS+=(-c "$CONFIG_FILE")
 fi
 
+# TODO: Remove this after all open alerts are categorized in the config files
+# Don't trigger a failure on WARNING level alerts, until we have a complete config file this will cause us to never
+# fail this step. However, if we do fail on warnings with the current configuration state every build will fail.
+# This is because ZAP defaults all alerts to WARN level, and any warnings found will trigger a failing exit code
+# from the zap-full-scan.py script.
+ZAP_ARGS+=(-I)
 
-# The `grep -q` piped to the end of the previous command will return a
-# 0 exit code if the term is found and 1 otherwise.
-ZAPEXIT=$?
+# Run the ZAP full scan and store output for further processing if needed.
+ZAP_OUTPUT=$(docker-compose run --rm zaproxy zap-full-scan.py "${ZAP_ARGS[@]}" | tee /dev/tty)
+ZAP_EXIT=$?
 
-if [ "$TARGET" = "frontend" ]; then
-    docker-compose down --remove-orphan
+if [ "$ZAP_EXIT" -eq 0 ]; then
+  echo "OWASP ZAP scan successful"
+else
+  echo "OWASP ZAP scan failed"
 fi
 
-EXIT=0
-
-if [ "$ZAPEXIT" = 1 ] ; then
-	echo "OWASP ZAP scan failed"
-	EXIT=1
+# Nightly scans in Circle CI need to persist some values across multiple steps.
+if [ "$ENVIRONMENT" = "nightly" ]; then
+  ZAP_SUMMARY=$(echo "$ZAP_OUTPUT" | tail -1 | xargs)
+
+  function get_summary_value () {
+    echo "$ZAP_SUMMARY" | grep -o "$1: [^ ]*" | cut -d':' -f2 | sed 's/[^0-9]*//g'
+  }
+
+  ZAP_PASS_COUNT=$(get_summary_value PASS)
+  ZAP_WARN_COUNT=$(get_summary_value WARN-NEW)
+  ZAP_FAIL_COUNT=$(get_summary_value FAIL-NEW)
+
+  TARGET=$(echo "$TARGET" | awk '{print toupper($0)}')
+  {
+    echo "export ZAP_${TARGET}_PASS_COUNT=$ZAP_PASS_COUNT"
+    echo "export ZAP_${TARGET}_WARN_COUNT=$ZAP_WARN_COUNT"
+    echo "export ZAP_${TARGET}_FAIL_COUNT=$ZAP_FAIL_COUNT"
+  } >> "$BASH_ENV"
```

Comment:
This makes these environment variables available to the later step that runs a CF Task with these values. 

Ref: https://circleci.com/docs/2.0/env-vars/#setting-an-environment-variable-in-a-shell-command